### PR TITLE
Blacklist and whitelist preprocessors

### PR DESCRIPTION
**Pull request by ostap0207**
This duplicates blacklist and whitelist in ruby logasm PR https://github.com/salemove/logasm/pull/18.

### CODE REVIEW

File: test/blacklist_test.coffee [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    it 'throws error', ->
+      expect(-> new Blacklist(config())).to.throw('Action: reverse is not supported')
+
+  context '#process', ->
+    beforeEach ->
+      blacklist = new Blacklist(config())
+      processedData = blacklist.process(data())
+
+    context 'when actions is "exclude"', ->
```
Comment (willmore) [inline test/blacklist_test.coffee]: actions -> action

File: test/blacklist_test.coffee [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        expect(processedData.array).to.be.instanceof(Array)
+
+      context 'when field is deeply nested', ->
+        depth = 10
+        data.is -> nest_data({}, depth, {field: value()})
+
+        it 'removes field', ->
+          expect(nested_field(processedData, depth)).to.not.include.keys('field')
+
+    context 'when actions is "mask"', ->
```
Comment (willmore) [inline test/blacklist_test.coffee]: actions -> action

**Comment by willmore** [comment]
Minor comments, otherwise +1.

File: package.json [hunk 3]
```diff
@@ -26,7 +26,9 @@
     "mocha": "latest",
     "chai": "latest",
     "should": ">=3.1",
-    "sinon": "latest"
+    "sinon": "latest",
```
Comment (urmastalimaa) [inline package.json:29]: Bump node-logasm version as well

File: test/logasm_test.coffee [hunk 4]
```diff
@@ -32,6 +33,30 @@ describe 'Logasm', ->
...
+    }
+
+    beforeEach ->
+      @logasm = new Logasm([adapter], [preprocessor])
+
+    it 'preprocesses data before logging', ->
+      @logasm.info('Received message', {data: 'data'})
+
+      expect(preprocessor.process).to.be.calledWith({data: 'data', message: 'Received message'})
+      expect(adapter.log).to.be.called
```
Comment (urmastalimaa) [inline test/logasm_test.coffee]: Assert that this is called with the output from the preprocessro

Comment (urmastalimaa) [inline test/logasm_test.coffee]: Assert that this is called with the output from the preprocessor

**Comment by urmastalimaa** [comment]
2 minor comments, +1

**Comment by urmastalimaa** [comment]
!squash

**Pull request merged**